From 4432c3bf580434c3ed16bea6b01acfef7d724799 Mon Sep 17 00:00:00 2001
From: Zidan Wang <zidan.wang@freescale.com>
Date: Thu, 2 Apr 2015 11:39:00 +0800
Subject: [PATCH] MLK-10556 ASoC: imx-wm8958: wm8958 can't support dac sample
 rate 64kHZ

wm8958 can't support dac sample rate 64kHZ, so add a constraint
list for it.

Signed-off-by: Zidan Wang <zidan.wang@freescale.com>
---
 sound/soc/fsl/imx-wm8958.c | 14 ++++++++++++++
 1 file changed, 14 insertions(+)

diff --git a/sound/soc/fsl/imx-wm8958.c b/sound/soc/fsl/imx-wm8958.c
index 544f8b27..fb0399c 100644
--- a/sound/soc/fsl/imx-wm8958.c
+++ b/sound/soc/fsl/imx-wm8958.c
@@ -228,16 +228,27 @@ static int imx_hifi_hw_free(struct snd_pcm_substream *substream)
 
 	return 0;
 }
+
 static u32 imx_wm8958_adc_rates[] = {
 	8000, 11025, 12000, 16000, 22050,
 	24000, 32000, 44100, 48000
 };
 
+static u32 imx_wm8958_dac_rates[] = {
+	8000, 11025, 12000, 16000, 22050,
+	24000, 32000, 44100, 48000, 88200, 96000
+};
+
 static struct snd_pcm_hw_constraint_list imx_wm8958_adc_rate_constraints = {
 	.count = ARRAY_SIZE(imx_wm8958_adc_rates),
 	.list = imx_wm8958_adc_rates,
 };
 
+static struct snd_pcm_hw_constraint_list imx_wm8958_dac_rate_constraints = {
+	.count = ARRAY_SIZE(imx_wm8958_dac_rates),
+	.list = imx_wm8958_dac_rates,
+};
+
 static int imx_hifi_startup(struct snd_pcm_substream *substream)
 {
 	struct snd_soc_pcm_runtime *rtd = substream->private_data;
@@ -256,6 +267,9 @@ static int imx_hifi_startup(struct snd_pcm_substream *substream)
 	if (substream->stream == SNDRV_PCM_STREAM_CAPTURE)
 		ret = snd_pcm_hw_constraint_list(substream->runtime, 0,
 			SNDRV_PCM_HW_PARAM_RATE, &imx_wm8958_adc_rate_constraints);
+	else if (substream->stream == SNDRV_PCM_STREAM_PLAYBACK)
+		ret = snd_pcm_hw_constraint_list(substream->runtime, 0,
+			SNDRV_PCM_HW_PARAM_RATE, &imx_wm8958_dac_rate_constraints);
 	return ret;
 }
 
-- 
1.8.0

